### ISSUE TITLE
cargo fix fails, having introduced compilation warnings

### ISSUE THREAD

**Issue by jimblandy** [rust-lang/rust#83315]
I tried to use `cargo fix --edition-idioms`, but it printed a lot of errors and declined to fix some things.

I've attached the output.
[output.txt](https://github.com/rust-lang/rust/files/6174862/output.txt)

### Meta

`rustc --version --verbose`:
```
cargo 1.52.0-nightly (90691f2bf 2021-03-16)
release: 1.52.0
commit-hash: 90691f2bfe9a50291a98983b1ed2feab51d5ca55
commit-date: 2021-03-16
```

**Comment by ehuss** [comment]
Hi!  I think this is a duplicate of #55768.  I believe the issue is that when a macro is invoked multiple times, rustc is issuing a warning for each invocation, and `cargo fix` is applying all the duplicate suggestions.

**Comment by fmease** [comment]
Closing as duplicate of #55768 which was fixed by #90446.
